mlflow/genai/scheduled_scorers.py

```diff
@@ -0,0 +1,393 @@
+    which helps track quality metrics like groundedness, safety, and guideline adherence
+    alongside operational metrics like volume, latency, and cost.
+
+    When configured, scheduled scorers run automatically in the background to evaluate
+    a sample of traces based on the specified sampling rate and filter criteria. The
+    Assessments are displayed in the Traces tab of the MLflow experiment and can be used to
+    identify quality issues in production.
+
+    Args:
+        scorer_fn: The scorer function to run on sampled traces. Must be either a built-in
+            scorer (e.g., Safety, Correctness) or a function decorated with @scorer.
+            Subclasses of Scorer are not supported.
+        scorer_name: A unique name for this scheduled scorer configuration.
+        sample_rate: The fraction of traces to evaluate, between 0.0 and 1.0. For example,
+            0.1 means 10% of traces will be randomly selected for evaluation.
+        filter_string: An optional MLflow search_traces compatible filter string to apply
+            before sampling traces. Only traces matching this filter will be considered
+            for evaluation. Uses the same syntax as mlflow.search_traces().
+
+    Example:
+        .. code-block:: python
+
+            from mlflow.genai.scorers import Safety, scorer
+            from mlflow.genai.scheduled_scorers import ScheduledScorer
+
+            # Using a built-in scorer
+            safety_config = ScheduledScorer(
+                scorer_fn=Safety(),
+                scorer_name="production_safety",
+                sample_rate=0.2,  # Evaluate 20% of traces
+                filter_string="trace.status = 'OK'",
+            )
+
+
+            # Using a custom scorer
+            @scorer
+            def response_length(outputs):
+                return len(str(outputs)) > 100
+
+
+            length_config = ScheduledScorer(
+                scorer_fn=response_length,
+                scorer_name="adequate_length",
+                sample_rate=0.1,  # Evaluate 10% of traces
+                filter_string="trace.status = 'OK'",
+            )
+
+    Note:
+        Scheduled scorers are executed automatically by Databricks and do not need to be
+        manually triggered. The Assessments appear in the Traces tab of the MLflow
+        experiment. Only traces logged directly to the experiment are monitored; traces
+        logged to individual runs within the experiment are not evaluated.
+    """
+
+    def __init__(
+        self,
+        *,
+        scorer_fn: Scorer,
+        scorer_name: str,
+        sample_rate: float,
+        filter_string: Optional[str] = None,
+    ):
+        self.scorer_fn = scorer_fn
+        self.scorer_name = scorer_name
+        self.sample_rate = sample_rate
+        self.filter_string = filter_string
+
+
+# Scheduled Scorer CRUD operations
+@experimental
+def add_scheduled_scorer(
+    *,
+    experiment_id: Optional[str] = None,
```

Comment:
fyi @nsthorat This API differs from what we'd previously discussed in 2 ways:
1. All parameters must be passed as keyword arguments. This gives us more flexibility to update the API in the future.
2. experiment_id is optional, and we use the active experiment_id if it's not set. This follows the convention of other MLflow Python APIs.